spec/models/poll_spec.rb

```diff
@@ -93,21 +111,187 @@
+      end
+    end
+
+    context 'when poll finished' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'finished' }
+
+      it 'have status finished' do
+        expect(poll).to be_finished
+      end
+
+      it 'able to be draft' do
+        expect(poll.able_to_draft?).to be_truthy
+      end
+
+      it 'able to be deleted' do
+        expect(poll.able_to_delete?).to be_truthy
+      end
+
+      it 'not able to be ready' do
+        expect(poll.able_to_ready?).to be_falsey
+      end
+    end
+
+    context 'when poll deleted' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'deleted' }
+
+      it 'have status deleted' do
+        expect(poll).to be_deleted
+      end
+
+      it 'able to be draft' do
+        expect(poll.able_to_draft?).to be_truthy
+      end
+
+      it 'not able to be ready' do
+        expect(poll.able_to_ready?).to be_falsey
+      end
+
+      it 'not able to be finished' do
+        expect(poll.able_to_finish?).to be_falsey
+      end
+    end
   end
 
-  it 'shold have error when reopen outdated' do
-    poll.update_attribute(:expire_at, 1.year.ago)
-    poll.closed!
+  context 'Check changing status callbacks.' do
+    context 'draft -> ready' do
+      let(:poll) { FactoryGirl.create :poll, :with_title }
+
+      it 'stay draft if poll have no options' do
+        poll.ready!
+
+        expect(poll).to be_draft
+      end
+
+      it 'have an error if poll have no options' do
+        expect { poll.ready! }.to change { poll.errors.count }.by(1)
+      end
+    end
+
+    context 'ready -> draft' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'ready' }
+      let(:poll2) { FactoryGirl.create :valid_poll, status: 'ready' }
+      before(:each) { poll.vote!(user, [0, 1, 2]) }
+      subject { poll.draft! }
 
-    expect(poll.errors[:expire_at]).to eq(['should be in future'])
+      context 'drop votation history' do
+        it 'have empty current state' do
+          subject
+
+          expect(poll.current_state).to be_empty
+        end
+
+        it 'have no vote results' do
+          subject
+
+          expect(poll.vote_results).to be_empty
+        end
+
+        it 'have no voters' do
+          subject
+
+          expect(poll.voters).to be_empty
+        end
+
+        it 'dissapear from users voted polls' do
+          poll2.vote!(user, [0, 1, 2])
+          subject
+
+          expect(user.voted_polls).to eq([poll2])
+        end
+      end
+    end
+
+    context 'ready -> finished' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'ready' }
+      subject { poll.finish! }
+
+      it 'keep votation progress' do
+        poll.vote!(user, [0, 1, 2])
+        subject
+
+        expect(poll.voters).to eq([user])
+      end
+    end
   end
 
-  it 'should close polls with expiration date in past' do
-    poll.update_attribute(:expire_at, 1.year.ago)
+  context 'draft poll when' do
+    before(:each) { poll.ready! }
```

Comment:
Maybe `let(:poll) { create(:valid_poll, :ready) }` will be better? It's possible (and recommended) to override your `let`s in nested blocks; and using factory traits instead of before blocks will make your tests faster (no additional roundtrip) and more controllable (if making poll ready would include more than one change afterwards, you'll have a single place to introduce that change — your factory).